spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -360,10 +399,73 @@ private static boolean hasNoInFilter(Predicate predicate) {
   }
 
   private static boolean isSupportedInPredicate(Predicate predicate) {
-    if (!isRef(childAtIndex(predicate, 0))) {
+    if (!couldConvert(childAtIndex(predicate, 0))) {
       return false;
     } else {
       return Arrays.stream(predicate.children()).skip(1).allMatch(SparkV2Filters::isLiteral);
     }
   }
+
+  /** Should be called after {@link #couldConvert} passed */
+  private static <T> UnboundTerm<Object> toTerm(T input) {
+    if (input instanceof NamedReference) {
+      return Expressions.ref(SparkUtil.toColumnName((NamedReference) input));
+    } else {
+      return udfToTerm((UserDefinedScalarFunc) input);
+    }
+  }
+
+  private static UnboundTerm<Object> udfToTerm(UserDefinedScalarFunc udf) {
+    switch (udf.name().toLowerCase(Locale.ROOT)) {
+      case "years":
+        Preconditions.checkArgument(
+            udf.children().length == 1, "years function should have only one children (column)");
+        Preconditions.checkArgument(
+            isRef(udf.children()[0]),
+            "The child of years function should be type of NamedReference");
```

Comment:
Error messages should be direct, rather than being phrased as a full sentence. In cases like this, I'd expect something like either "Invalid arguments: expected one column name but was %s" or "Cannot convert UDF: expected one column argument, but was %s". In any case, there should not be an exception thrown so it doesn't matter here.

Reply:
Thanks for the good advice.